addons/payment_authorize/models/payment.py

```diff
@@ -48,13 +48,36 @@ def _get_authorize_urls(self, environment):
             return {'authorize_form_url': 'https://test.authorize.net/gateway/transact.dll'}
 
     def _authorize_generate_hashing(self, values):
+        def is_signature_key(key):
+            ''' Try to detect if the key is new hexa signature key or the old
+            transaction key. Basically this method checks if the user proceeded
+            to the key change required from Authorize.net the 28th March 2019.
+
+            Transaction key (old): 16 length, anything between 0-9, a-z and A-Z.
+            Signature key (new): 128 length, hexa, only 0-9 and A-Z.
+            '''
+            try:
+                int(key, 16)
```

Comment:
use all(c in string.hexdigits for c in key) an remove try catch

Reply:
Much cleaner to get rid of the try/except indeed.
Ended up merging both lines:
```python
not_hexa = [c for c in key if c.islower() or c not in string.hexdigits]
```

Reply:
Btw, `any` and `all` will indeed [short-circuit the execution (note 2)](https://stackoverflow.com/a/19389957/3974585).

So I also updated the
```python
[c for c in key if c.islower()]
```
to
```python
any(c.islower() for c in key)
```